### ISSUE TITLE
Metric Calculation

### ISSUE THREAD

**Issue by 647-bei** [House-yuyu/AFD-former#5]
I downloaded the dataset you provided, but the calculated PSNR and other metrics differ from those in your paper. Could you please clarify which codebase you used to compute these evaluation metrics?

**Comment by House-yuyu** [comment]
Hi @647-bei, apologies for my delayed response. I have been quite busy recently.

1. You should download the pre-trained model weights from Baidu Cloud Drive or Google Cloud Drive.
2. Calculating the evaluation metrics through the Y channel.

if it still doesn't work, please feel free to contact.

**Comment by 647-bei** [comment]
Yes, I calculated it on the Y channel, but for example, the PSNR and SSIM IWSSIM calculated by different library versions have different values, so can you tell me which version of the calculation method is used or what do you implement yourself? What kind of calculation?

**Comment by 647-bei** [comment]
> Yes, I calculated it on the Y channel, but for example, the PSNR and SSIM IWSSIM calculated by different library versions have different values, so can you tell me which version of the calculation method is used or what do you implement yourself? What kind of calculation?

For example, when I calculate PSNR and SSIM, I call cv2.PSNR, so the calculated indicators are different from the indicators of the test set provided in your paper.

**Comment by 647-bei** [comment]
> > Yes, I calculated it on the Y channel, but for example, the PSNR and SSIM IWSSIM calculated by different library versions have different values, so can you tell me which version of the calculation method is used or what do you implement yourself? What kind of calculation?
> 
> For example, when I calculate PSNR and SSIM, I call cv2.PSNR, so the calculated indicators are different from the indicators of the test set provided in your paper.

For example, on the H264 Kendo sequence of the Testing datasets, the psnr I calculated is 33.54, the SSIM is 0.9224, and the paper provides 33.57/0.949

**Comment by House-yuyu** [comment]
Oh, I forgot to release the evaluation code, I'll upload it soon.

**Comment by 647-bei** [comment]
> Oh, I forgot to release the evaluation code, I'll upload it soon.

Thank you for your help. I plan to follow your work, so I need to use your previous comparative experiments. wish you a happy life

**Comment by 647-bei** [comment]
> > Oh, I forgot to release the evaluation code, I'll upload it soon.
> 
> Thank you for your help. I plan to follow your work, so I need to use your previous comparative experiments. wish you a happy life

Haha, so I found some errors when calculating the indicator

**Comment by House-yuyu** [comment]
@647-bei, hi, the evaluation code has been updated, please take care to change the relevant path settings, etc.

**Comment by 647-bei** [comment]
> @647-bei, hi, the evaluation code has been updated, please take care to change the relevant path settings, etc.

Thank you. Do you have implementations for other evaluation metrics mentioned in your paper? For example, in calculating MP-PSNR, how did you choose to perform the weighting – is it uniform weighting?

**Comment by House-yuyu** [comment]
> > @647-bei, hi, the evaluation code has been updated, please take care to change the relevant path settings, etc.
> 
> Thank you. Do you have implementations for other evaluation metrics mentioned in your paper? For example, in calculating MP-PSNR, how did you choose to perform the weighting – is it uniform weighting?

Apologies for forgetting to upload the rest of the code last time. The other metrics like MP-PSNR and SC-IQA are MATLAB based code, and I'll get those up soon!

**Comment by 647-bei** [comment]
Thank you. Wish you a Happy New Year!

**Comment by 647-bei** [comment]
> > > @647-bei, hi, the evaluation code has been updated, please take care to change the relevant path settings, etc.
> > 
> > 
> > Thank you. Do you have implementations for other evaluation metrics mentioned in your paper? For example, in calculating MP-PSNR, how did you choose to perform the weighting – is it uniform weighting?
> 
> Apologies for forgetting to upload the rest of the code last time. The other metrics like MP-PSNR and SC-IQA are MATLAB based code, and I'll get those up soon!

Thank you.Wish you a Happy New Year!

**Comment by House-yuyu** [comment]
Hi, @647-bei, please download the MATLAB-based evaluation code in Google Drive (same address as the pre-trained model weights). I didn't provide a download link to Baidu cloud disk since there was an error while uploading and didn't succeed in solving it.

**Comment by 647-bei** [comment]
> Google Drive

Thank you, but you need to apply for permission to access Google Drive. I submitted an access application. Please go through it.

**Comment by 647-bei** [comment]
> 您好，请在Google Drive中下载基于MATLAB的评估代码（与预训练模型权重的地址相同）。我没有提供百度云盘的下载链接，因为上传时出现了错误，没有成功解决。

Thanks, but you need to apply for permission to access Google Drive. I submitted an access application.

**Comment by 647-bei** [comment]
> 您好，请在Google Drive中下载基于MATLAB的评估代码（与预训练模型权重的地址相同）。我没有提供百度云盘的下载链接，因为上传时出现了错误，没有成功解决。

Thanks, but your Google Drive settings require permission to access. I submitted an access request.

**Comment by House-yuyu** [comment]
I solved the permissions problem. Please try it again.
